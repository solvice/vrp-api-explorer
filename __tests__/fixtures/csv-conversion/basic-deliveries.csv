name,latitude,longitude,duration,priority,capacity_needed
London Office,51.5074,0.1278,30,high,100
Manchester Store,53.4808,-2.2426,45,medium,150
Birmingham Warehouse,52.4862,-1.8904,25,low,75