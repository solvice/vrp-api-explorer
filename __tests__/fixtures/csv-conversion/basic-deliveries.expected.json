{
  "vrpData": {
    "jobs": [
      {
        "id": "job_1",
        "name": "London Office",
        "location": {
          "latitude": 51.5074,
          "longitude": 0.1278
        },
        "duration": 1800,
        "priority": "high",
        "demand": [100]
      },
      {
        "id": "job_2",
        "name": "Manchester Store",
        "location": {
          "latitude": 53.4808,
          "longitude": -2.2426
        },
        "duration": 2700,
        "priority": "medium",
        "demand": [150]
      },
      {
        "id": "job_3",
        "name": "Birmingham Warehouse",
        "location": {
          "latitude": 52.4862,
          "longitude": -1.8904
        },
        "duration": 1500,
        "priority": "low",
        "demand": [75]
      }
    ],
    "resources": [
      {
        "id": "vehicle_1",
        "capacity": [425],
        "shifts": [
          {
            "start_location": {
              "latitude": 51.5074,
              "longitude": 0.1278
            },
            "end_location": {
              "latitude": 51.5074,
              "longitude": 0.1278
            }
          }
        ]
      }
    ]
  },
  "explanation": "Converted 3 locations to VRP jobs with duration conversion from minutes to seconds",
  "conversionNotes": [
    "Duration converted from minutes to seconds",
    "Priority mapped to job priority field",
    "Capacity needs mapped to demand array"
  ],
  "rowsProcessed": 3
}